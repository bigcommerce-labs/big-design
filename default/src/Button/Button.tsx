import React from 'react';
import styled from 'styled-components';

import { defaultTheme, ThemeInterface } from '../themes/basic';
import { Spinner } from '../Spinner/Spinner';

import { ButtonStyles } from './styles';

interface Props {
  actionType: 'normal' | 'destructive';
  icon?: React.ReactChild;
  iconLeft?: React.ReactChild;
  iconRight?: React.ReactChild;
  isLoading: boolean;
  spinner: React.ReactChild;
  theme: ThemeInterface;
  variant: 'primary' | 'secondary' | 'subtle';
}

export type ButtonProps = Props & React.ButtonHTMLAttributes<HTMLButtonElement>;

export const StyledButton = styled('button')<ButtonProps>`
  ${({ theme }) => theme.Button || ButtonStyles};
`;

export class Button extends React.PureComponent<ButtonProps> {
  static defaultProps: Partial<ButtonProps> = {
    actionType: 'normal',
    isLoading: false,
    spinner: <Spinner overlay={false} />,
    theme: defaultTheme,
    variant: 'primary',
  };

  render() {
    const isLoading = this.props.isLoading;

    return (
      <StyledButton role="button" tabIndex={0} {...this.props} onClick={this.handleClick}>
        {isLoading ? this.props.spinner : null}
        <ContentWrapper isLoading={isLoading} theme={this.props.theme}>
          {this.props.iconLeft}
          {this.props.icon}
          {!this.props.icon && this.props.children}
          {this.props.iconRight}
        </ContentWrapper>
      </StyledButton>
    );
  }

  private handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
    const { disabled, isLoading } = this.props;

    if (this.props.onClick && !disabled && !isLoading) {
      this.props.onClick(event);
    }
  };
}

const ContentWrapper = styled.span.attrs<{}, { isLoading?: boolean }>({})`
  align-content: center;
  align-items: center;
  display: inline-grid;
  grid-auto-flow: column;
  grid-gap: ${({ theme }) => theme.spacing.xSmall};
  visibility: ${({ isLoading }) => (isLoading ? 'hidden' : 'visible')};
`;
