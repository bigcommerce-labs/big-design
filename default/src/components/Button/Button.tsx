import React from 'react';
import styled from 'styled-components';

import { defaultTheme, ThemeInterface } from '../../theme';
import { Spinner } from '../Spinner';

import { ButtonContentWrapperStyles, ButtonStyles } from './styles';

interface Props {
  actionType: 'normal' | 'destructive';
  iconOnly?: React.ReactChild;
  iconLeft?: React.ReactChild;
  iconRight?: React.ReactChild;
  isLoading: boolean;
  spinner: React.ReactChild;
  theme?: ThemeInterface;
  variant: 'primary' | 'secondary' | 'subtle';
}

export type ButtonProps = Props & React.ButtonHTMLAttributes<HTMLButtonElement>;

export class Button extends React.PureComponent<ButtonProps> {
  static defaultProps: Partial<ButtonProps> = {
    actionType: 'normal',
    isLoading: false,
    spinner: <Spinner overlay={false} />,
    variant: 'primary',
  };

  render() {
    const isLoading = this.props.isLoading;

    return (
      <StyledButton role="button" tabIndex={0} {...this.props} onClick={this.handleClick}>
        {isLoading ? this.props.spinner : null}
        <ContentWrapper isLoading={isLoading} theme={this.props.theme}>
          {this.props.iconLeft}
          {this.props.iconOnly}
          {!this.props.iconOnly && this.props.children}
          {this.props.iconRight}
        </ContentWrapper>
      </StyledButton>
    );
  }

  private handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
    const { disabled, isLoading } = this.props;

    if (this.props.onClick && !disabled && !isLoading) {
      this.props.onClick(event);
    }
  };
}

export const StyledButton = styled('button')<ButtonProps>`
  ${({ theme }) => theme.Button || ButtonStyles};
`;

const ContentWrapper = styled.span.attrs<{}, { isLoading?: boolean }>({})`
  ${({ theme }) => theme.ButtonContentWrapperStyles || ButtonContentWrapperStyles};
`;

StyledButton.defaultProps = { theme: defaultTheme };
ContentWrapper.defaultProps = { theme: defaultTheme };
