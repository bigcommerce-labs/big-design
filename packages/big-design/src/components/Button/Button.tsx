import React, { ComponentPropsWithoutRef, forwardRef, memo, Ref } from 'react';

import { MarginProps } from '../../mixins';
import { ProgressCircle } from '../ProgressCircle';

import { ContentWrapper, LoadingSpinnerWrapper, StyledButton } from './styled';

interface PrivateProps {
  forwardedRef: Ref<HTMLButtonElement>;
}

export interface ButtonProps extends ComponentPropsWithoutRef<'button'>, MarginProps {
  actionType?: 'normal' | 'destructive';
  children?: React.ReactNode;
  iconLeft?: React.ReactNode;
  iconOnly?: React.ReactNode;
  iconRight?: React.ReactNode;
  isLoading?: boolean;
  mobileWidth?: 'auto' | '100%';
  variant?: 'primary' | 'secondary' | 'subtle' | 'utility';
}

const LoadingSpinner = () => (
  <LoadingSpinnerWrapper alignItems="center">
    <ProgressCircle size="xxSmall" />
  </LoadingSpinnerWrapper>
);

const RawButton: React.FC<ButtonProps & PrivateProps> = memo(
  ({ forwardedRef, isLoading, disabled, ...props }) => {
    return (
      <StyledButton
        className="bd-button"
        {...props}
        disabled={isLoading || disabled}
        ref={forwardedRef}
      >
        {isLoading ? <LoadingSpinner /> : null}
        <ContentWrapper isLoading={isLoading}>
          {!props.iconOnly && props.iconLeft}
          {props.iconOnly}
          {!props.iconOnly && props.children}
          {!props.iconOnly && props.iconRight}
        </ContentWrapper>
      </StyledButton>
    );
  },
);

export const StyleableButton = forwardRef<HTMLButtonElement, ButtonProps>((props, ref) => (
  <RawButton {...props} forwardedRef={ref} />
));

export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
  ({ className, style, ...props }, ref) => <RawButton {...props} forwardedRef={ref} />,
);

const defaultProps = {
  actionType: 'normal' as const,
  isLoading: false,
  mobileWidth: '100%' as const,
  variant: 'primary' as const,
};

Button.displayName = 'Button';
Button.defaultProps = { ...defaultProps };

StyleableButton.displayName = 'StyleableButton';
StyleableButton.defaultProps = { ...defaultProps };
