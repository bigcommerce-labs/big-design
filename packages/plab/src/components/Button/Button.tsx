import React, { Ref } from 'react';

import { MarginProps } from '../../mixins';
import { ThemeInterface } from '../../theme';
import { Spinner } from '../Spinner';

import { ContentWrapper, StyledButton } from './styled';

interface PrivateProps {
  forwardedRef: Ref<HTMLButtonElement>;
}

export interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement>, MarginProps {
  actionType?: 'normal' | 'destructive';
  iconLeft?: React.ReactChild;
  iconOnly?: React.ReactChild;
  iconRight?: React.ReactChild;
  isLoading?: boolean;
  spinner?: React.ReactChild;
  theme?: ThemeInterface;
  variant?: 'primary' | 'secondary' | 'subtle';
}

class RawButton extends React.PureComponent<ButtonProps & PrivateProps> {
  render() {
    const { forwardedRef, ...props } = this.props;

    return (
      <StyledButton role="button" tabIndex={0} {...props} onClick={this.handleClick} ref={forwardedRef}>
        {props.isLoading ? props.spinner : null}
        <ContentWrapper isLoading={props.isLoading} theme={props.theme}>
          {props.iconLeft}
          {props.iconOnly}
          {!props.iconOnly && props.children}
          {props.iconRight}
        </ContentWrapper>
      </StyledButton>
    );
  }

  private handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
    const { disabled, isLoading } = this.props;

    if (this.props.onClick && !disabled && !isLoading) {
      this.props.onClick(event);
    }
  };
}

export const StyleableButton = React.forwardRef<HTMLButtonElement, ButtonProps>((props, ref) => (
  <RawButton {...props} forwardedRef={ref} />
));

export const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(({ className, style, ...props }, ref) => (
  <RawButton {...props} forwardedRef={ref} />
));

const defaultProps = {
  actionType: 'normal' as 'normal',
  isLoading: false,
  spinner: <Spinner overlay={false} />,
  variant: 'primary' as 'primary',
};

Button.displayName = 'Button';
Button.defaultProps = { ...defaultProps };

StyleableButton.displayName = 'StyleableButton';
StyleableButton.defaultProps = { ...defaultProps };
