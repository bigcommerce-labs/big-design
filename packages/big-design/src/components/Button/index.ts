import { ButtonProps as _ButtonProps } from './Button';

export { Button } from './Button';
export type ButtonProps<T extends 'button' | 'a'> = _ButtonProps<T>;
