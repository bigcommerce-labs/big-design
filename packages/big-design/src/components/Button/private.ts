import { theme as defaultTheme } from '@bigcommerce/big-design-theme';
import styled from 'styled-components';

import { ButtonProps, StyleableButton as _StyleableButton } from './Button';

export const StyleableButton = _StyleableButton;

export const MessagingButton = styled(_StyleableButton)<ButtonProps>`
  background-color: transparent;
  border: ${({ theme }) => theme.border.none};
  color: ${({ theme }) => theme.colors.secondary70};
  height: auto;
  padding: ${({ theme }) => theme.spacing.none};

  &:active {
    background-color: ${({ theme }) => theme.colors.primary20};
  }

  &:focus {
    box-shadow: ${({ theme }) => `0 0 0 ${theme.spacing.xxSmall} ${theme.colors.primary20}`};
  }

  &:hover:not(:active) {
    background-color: ${({ theme }) => theme.colors.primary10};
  }

  &[disabled] {
    color: ${({ theme }) => theme.colors.secondary30};
  }
`;

MessagingButton.defaultProps = { theme: defaultTheme };
