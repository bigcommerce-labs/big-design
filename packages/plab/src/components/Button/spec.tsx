import 'jest-styled-components';
import React from 'react';
import { cleanup, render } from 'react-testing-library';

import { Button } from './Button';

afterEach(cleanup);

test('render default button', () => {
  const { container } = render(<Button>Button</Button>);

  expect(container.firstChild).toMatchSnapshot();
});

test('render disabled button', () => {
  const { container } = render(<Button disabled>Button</Button>);

  expect(container.firstChild).toMatchSnapshot();
});

test('render destructive button', () => {
  const { container } = render(<Button actionType="destructive">Button</Button>);

  expect(container.firstChild).toMatchSnapshot();
});

test('render destructive disabled button', () => {
  const { container } = render(
    <Button actionType="destructive" disabled>
      Button
    </Button>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('render secondary button', () => {
  const { container } = render(<Button variant="secondary">Button</Button>);

  expect(container.firstChild).toMatchSnapshot();
});

test('render secondary disabled button', () => {
  const { container } = render(
    <Button variant="secondary" disabled>
      Button
    </Button>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('render secondary destructive button', () => {
  const { container } = render(
    <Button variant="secondary" actionType="destructive">
      Button
    </Button>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('render secondary destructive disabled button', () => {
  const { container } = render(
    <Button variant="secondary" actionType="destructive" disabled>
      Button
    </Button>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('render subtle button', () => {
  const { container } = render(<Button variant="subtle">Button</Button>);

  expect(container.firstChild).toMatchSnapshot();
});

test('render subtle disabled button', () => {
  const { container } = render(
    <Button variant="subtle" disabled>
      Button
    </Button>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('render subtle destructive button', () => {
  const { container } = render(
    <Button variant="subtle" actionType="destructive">
      Button
    </Button>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('render subtle destructive disabled button', () => {
  const { container } = render(
    <Button variant="subtle" actionType="destructive" disabled>
      Button
    </Button>,
  );

  expect(container.firstChild).toMatchSnapshot();
});
