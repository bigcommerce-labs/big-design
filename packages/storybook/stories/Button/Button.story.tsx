import { Button, DropdownIcon, Grid, H0, H1, H2, Link, PlusIcon, Text } from '@bigcommerce/big-design';
import { storiesOf } from '@storybook/react';
import React from 'react';

import { Code, CodePreview } from '../../components';

import { ButtonPropTable } from './';

storiesOf('Button', module).add('Overview', () => (
  <>
    <H0>Buttons</H0>

    <Text>
      Buttons are calls to action used throughout the product experience.{' '}
      <Link href="https://bigcommerce.design/buttons" target="_blank">
        Buttons Design Guidelines
      </Link>
    </Text>

    <CodePreview>
      {/* jsx-to-string:start */}
      <Button actionType="normal" isLoading={false} variant="primary">
        Label
      </Button>
      {/* jsx-to-string:end */}
    </CodePreview>

    <H1>API</H1>

    <ButtonPropTable />

    <H1>Variants</H1>

    <Text>
      There are three types of variants to choose from: <Code>primary</Code>, <Code>secondary</Code>, and{' '}
      <Code>subtle</Code>. You can determine what type of variant by using the <Code primary>variant</Code> prop.
    </Text>

    <CodePreview>
      {/* jsx-to-string:start */}
      <Grid columns="repeat(3, min-content)">
        <Button variant="primary">Primary</Button>
        <Button variant="secondary">Secondary</Button>
        <Button variant="subtle">Subtle</Button>
      </Grid>
      {/* jsx-to-string:end */}
    </CodePreview>

    <H1>Action Types</H1>

    <Text>
      There are two action types: <Code>normal</Code> &amp; <Code>destructive</Code>. They are used to indicate the
      nature of the action when clicking on the button.
    </Text>

    <CodePreview>
      {/* jsx-to-string:start */}
      <Grid columns="repeat(2, min-content)">
        <Button actionType="normal">Normal</Button>
        <Button actionType="destructive">Destructive</Button>
      </Grid>
      {/* jsx-to-string:end */}
    </CodePreview>

    <H1>States</H1>

    <H2>Loading</H2>

    <Text>The loading state is used when clicking a button will perform a asyncronous action.</Text>

    <CodePreview>
      {/* jsx-to-string:start */}
      {function LoadingButton() {
        const [isLoading, setLoading] = React.useState(false);

        function simulateNetworkRequest() {
          return new Promise(resolve => setTimeout(resolve, 2000));
        }

        React.useEffect(() => {
          if (isLoading) {
            simulateNetworkRequest().then(() => {
              setLoading(false);
            });
          }
        }, [isLoading]);

        const handleClick = () => setLoading(true);

        return (
          <Button variant="primary" isLoading={isLoading} onClick={!isLoading ? handleClick : () => null}>
            Click to load
          </Button>
        );
      }}
      {/* jsx-to-string:end */}
    </CodePreview>

    <H2>Disabled</H2>

    <Text>
      A disabled state is used to indicate no action can be done using the button by passing a{' '}
      <Code primary>disabled</Code> prop.
    </Text>

    <CodePreview>
      {/* jsx-to-string:start */}
      <Grid columns="repeat(3, min-content)">
        <Button disabled>Disabled</Button>
        <Button disabled variant="secondary">
          Disabled
        </Button>
        <Button disabled variant="subtle">
          Disabled
        </Button>
      </Grid>
      {/* jsx-to-string:end */}
    </CodePreview>

    <H1>Icons</H1>

    <Text>
      A button can also include icons on either side of the text (or both). When using <Code primary>iconOnly</Code>,
      the <Code primary>iconLeft</Code> &amp; <Code primary>iconRight</Code> components will be removed.
    </Text>

    <CodePreview>
      {/* jsx-to-string:start */}
      <Grid columns="repeat(4, min-content)">
        <Button iconOnly={<PlusIcon title="add" />} />
        <Button iconLeft={<PlusIcon />}>Label</Button>
        <Button iconLeft={<PlusIcon />} iconRight={<DropdownIcon />}>
          Label
        </Button>
        <Button iconRight={<DropdownIcon />}>Label</Button>
      </Grid>
      {/* jsx-to-string:end */}
    </CodePreview>
  </>
));
