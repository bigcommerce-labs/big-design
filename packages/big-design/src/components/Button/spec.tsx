import 'jest-styled-components';
import React from 'react';
import { fireEvent, render } from 'react-testing-library';

import { PlusIcon } from '../Icons';

import { Button } from './index';
import { StyleableButton } from './private';

test('render default button', () => {
  const { container } = render(<Button>Button</Button>);

  expect(container.firstChild).toMatchSnapshot();
});

test('render disabled button', () => {
  const { container } = render(<Button disabled>Button</Button>);

  expect(container.firstChild).toMatchSnapshot();
});

test('render destructive button', () => {
  const { container } = render(<Button actionType="destructive">Button</Button>);

  expect(container.firstChild).toMatchSnapshot();
});

test('render destructive disabled button', () => {
  const { container } = render(
    <Button actionType="destructive" disabled>
      Button
    </Button>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('render secondary button', () => {
  const { container } = render(<Button variant="secondary">Button</Button>);

  expect(container.firstChild).toMatchSnapshot();
});

test('render secondary disabled button', () => {
  const { container } = render(
    <Button variant="secondary" disabled>
      Button
    </Button>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('render secondary destructive button', () => {
  const { container } = render(
    <Button variant="secondary" actionType="destructive">
      Button
    </Button>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('render secondary destructive disabled button', () => {
  const { container } = render(
    <Button variant="secondary" actionType="destructive" disabled>
      Button
    </Button>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('render subtle button', () => {
  const { container } = render(<Button variant="subtle">Button</Button>);

  expect(container.firstChild).toMatchSnapshot();
});

test('render subtle disabled button', () => {
  const { container } = render(
    <Button variant="subtle" disabled>
      Button
    </Button>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('render subtle destructive button', () => {
  const { container } = render(
    <Button variant="subtle" actionType="destructive">
      Button
    </Button>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('render subtle destructive disabled button', () => {
  const { container } = render(
    <Button variant="subtle" actionType="destructive" disabled>
      Button
    </Button>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('render loading button', () => {
  const { container } = render(<Button isLoading={true}>Button</Button>);

  expect(container.firstChild).toMatchSnapshot();
});

test('render icon only button', () => {
  const { container } = render(<Button iconOnly={<PlusIcon />}>Button</Button>);

  expect(container.firstChild).toMatchSnapshot();
});

test('render icon left button', () => {
  const { container } = render(<Button iconLeft={<PlusIcon />}>Button</Button>);

  expect(container.firstChild).toMatchSnapshot();
});

test('render icon right button', () => {
  const { container } = render(<Button iconRight={<PlusIcon />}>Button</Button>);

  expect(container.firstChild).toMatchSnapshot();
});

test('render icon left and right button', () => {
  const { container } = render(
    <Button iconLeft={<PlusIcon />} iconRight={<PlusIcon />}>
      Button
    </Button>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('forwards ref', () => {
  const ref = React.createRef<HTMLButtonElement>();

  const { container } = render(<Button ref={ref} />);
  const button = container.querySelector('button');

  expect(button).toBe(ref.current);
});

test('triggers onClick', () => {
  const onClick = jest.fn();

  const { container } = render(<Button onClick={onClick} />);
  const button = container.firstChild as HTMLElement;

  fireEvent.click(button);

  expect(onClick).toHaveBeenCalled();
});

test('does not forward styles', () => {
  const { container } = render(<Button className="test" style={{ background: 'red' }} />);

  expect(container.getElementsByClassName('test').length).toBe(0);
  expect(container.firstChild).not.toHaveStyle('background: red');
});

test('private StyleableButton forwards styles', () => {
  const { container } = render(<StyleableButton className="test" style={{ background: 'red' }} />);

  expect(container.getElementsByClassName('test').length).toBe(1);
  expect(container.firstChild).toHaveStyle('background: red');
});

test('render only icon only with left and right icons button', () => {
  const plusIcon = <PlusIcon data-testid="icon-only" />;
  const { getAllByTestId } = render(
    <Button iconLeft={plusIcon} iconOnly={plusIcon} iconRight={plusIcon}>
      Button
    </Button>,
  );

  expect(getAllByTestId('icon-only').length).toBe(1);
});
