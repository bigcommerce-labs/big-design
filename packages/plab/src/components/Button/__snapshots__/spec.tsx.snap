// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sample test 1`] = `
.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border-radius: 0.25rem;
  border-style: solid;
  border-width: 1px;
  color: #FFFFFF;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  font-weight: 400;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 2.25rem;
  line-height: 2rem;
  outline: none;
  padding: 0 1rem;
  pointer-events: auto;
  position: relative;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
  width: 100%;
  background-color: #3C64F4;
  border-color: #3C64F4;
  font-weight: 600;
}

.c1:focus {
  outline: none;
}

.c1[disabled] {
  border-color: #D9DCE9;
  pointer-events: none;
}

.c1 + .c0 {
  margin-left: 1rem;
}

.c1:active {
  background-color: #0B38D9;
}

.c1:focus {
  box-shadow: 0 0 0 4px #DBE3FE;
}

.c1:hover:not(:active) {
  background-color: #2852EB;
}

.c1[disabled] {
  background-color: #D9DCE9;
}

.c2 {
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: inline-grid;
  grid-auto-flow: column;
  grid-gap: 0.5rem;
  visibility: visible;
}

@media (min-width:720px) {
  .c1 {
    width: auto;
  }
}

<button
  className="c0 c1"
  onClick={[Function]}
  role="button"
  tabIndex={0}
>
  <span
    className="c2"
  />
</button>
`;
